server/src/main/java/org/apache/druid/metadata/IndexerSQLMetadataStorageCoordinator.java

```diff
@@ -2054,18 +2054,37 @@ protected DataStoreMetadataUpdateResult updateDataSourceMetadataWithHandle(
     }
 
     final boolean startMetadataMatchesExisting;
+    int startMetadataGreaterThanExisting = 0;
```

Comment:
Can this be a boolean instead?

Reply:
Yes, we can have a boolean. I have implemented the Comparable for compareTo() method and it returns int. so, I didn't change it.

Reply:
`comareTo()` function returns +1, -1 and 0 for greaterThan, lessThan and Equal repectively.